samples/docScanner/js/docProcessing.js

```diff
@@ -0,0 +1,399 @@
+
+  // Extract image data.
+  cv.cvtColor(image, image, cv.COLOR_GRAY2RGBA);
+  let imgData = new ImageData(new Uint8ClampedArray(image.data),
+    image.cols, image.rows);
+
+  let startX = (video.width - image.cols) / 2;
+  let startY = (video.height - image.rows) / 2;
+  canvasContext.putImageData(imgData, startX, startY);
+}
+
+function fourPointTransform(image, rect, warpedImage) {
+  // Compute the width of the new image, which will be the
+  // maximum distance between bottom-right and bottom-left
+  // x-coordiates or the top-right and top-left x-coordinates.
+  let widthA = Math.sqrt(Math.pow(
+    rect[2].x - rect[3].x, 2) + Math.pow(rect[2].y - rect[3].y, 2));
+  let widthB = Math.sqrt(Math.pow(
+    rect[1].x - rect[0].x, 2) + Math.pow(rect[1].y - rect[0].y, 2));
+  let maxWidth = parseInt(Math.max(widthA, widthB));
+
+  // Compute the height of the new image, which will be the
+  // maximum distance between the top-right and bottom-right
+  // y-coordinates or the top-left and bottom-left y-coordinates.
+  let heightA = Math.sqrt(Math.pow(
+    rect[1].x - rect[2].x, 2) + Math.pow(rect[1].y - rect[2].y, 2));
+  let heightB = Math.sqrt(Math.pow(
+    rect[0].x - rect[3].x, 2) + Math.pow(rect[0].y - rect[3].y, 2));
+  let maxHeight = parseInt(Math.max(heightA, heightB));
+
+  // Now that we have the dimensions of the new image, construct
+  // the set of destination points to obtain a "birds eye view",
+  // (i.e. top-down view) of the image, again specifying points
+  // in the top-left, top-right, bottom-right, and bottom-left order.
+  let dsize = new cv.Size(maxWidth, maxHeight);
+  let imgTri = cv.matFromArray(4, 1, cv.CV_32FC2,
+    [rect[0].x, rect[0].y, rect[1].x, rect[1].y, rect[2].x, rect[2].y, rect[3].x, rect[3].y]);
+  let dstTri = cv.matFromArray(4, 1, cv.CV_32FC2,
+    [0, 0, maxWidth - 1, 0, maxWidth - 1, maxHeight - 1, 0, maxHeight - 1]);
+  let M = cv.getPerspectiveTransform(imgTri, dstTri);
+  cv.warpPerspective(image, warpedImage, M, dsize);
+
+  M.delete(); imgTri.delete(); dstTri.delete();
+}
+
+function drawPoints() {
+  cv.imshow('canvasOutput', src);
+
+  canvasContext.strokeStyle = '#57CC65';
+  for (let i = 0; i < approxCoords.length; i++) {
+    canvasContext.beginPath();
+    canvasContext.arc(approxCoords[i].x, approxCoords[i].y, 5, 0, 2 * Math.PI, false);
+    canvasContext.lineWidth = 10;
+    canvasContext.stroke();
+  }
+  canvasContext.beginPath();
+  canvasContext.moveTo(approxCoords[0].x, approxCoords[0].y);
+  canvasContext.lineTo(approxCoords[1].x, approxCoords[1].y);
+  canvasContext.lineTo(approxCoords[2].x, approxCoords[2].y);
+  canvasContext.lineTo(approxCoords[3].x, approxCoords[3].y);
+  canvasContext.lineTo(approxCoords[0].x, approxCoords[0].y);
+  canvasContext.lineWidth = 3;
+  canvasContext.stroke();
+}
+
+function addCanvasEventListeners() {
+  if (isMobileDevice()) {
+    canvasOutput.addEventListener('touchstart', startDragging);
```

Comment:
TODO : later on we can try to see if we can minimize the lines of code for adding/removing the listeners.